### PR TITLE
chore(account-settings): overrides cleanup

### PR DESCRIPTION
**Pull request by wlee221**
#### Description of changes

Cleans up component overrides:
- Mark essential override props as required (those that are always passed by parent control component and are needed for component to work)
- Rename `DeleteUser.SubmitButton` -> `DeleteUser.DeleteButton`, because `DeleteUser` isn't really a form and its button is just a plain button, not a form submit button.

#### Description of how you validated changes

Updated unit tests

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/src/components/AccountSettings/types.ts [hunk 1]
```diff
@@ -6,29 +6,42 @@ import {
...
+ * Any essential props for overriding components are marked as required.
+ */
+type CommonPasswordFieldProps = Partial<PasswordFieldPrimitiveProps> &
+  Required<Pick<PasswordFieldPrimitiveProps, 'onBlur' | 'onChange'>>;
+type CommonButtonProps = Partial<ButtonPrimtiveProps> &
+  Required<Pick<ButtonPrimtiveProps, 'onClick'>>;
+type CommonSubmitButtonProps = Partial<PrimitiveProps<ButtonProps, 'button'>>;
 type CommonAlertProps = Partial<PrimitiveProps<AlertProps, 'div'>>;
 
 /**
```
Comment (wlee221) [inline packages/react/src/components/AccountSettings/types.ts:29]:
```suggestion
/*
```

**Review by dbanksdesign — APPROVED** [review]

File: packages/react/src/components/AccountSettings/types.ts [hunk 2]
```diff
@@ -6,29 +6,45 @@ import {
...
  *
  * Note that `PrimitiveProps` is used to get native html types, like `onSubmit`.
  */
-type CommonPasswordFieldProps = Partial<
-  PrimitiveProps<PasswordFieldProps, 'input'>
->;
-type CommonButtonProps = Partial<PrimitiveProps<ButtonProps, 'button'>>;
-type CommonAlertProps = Partial<PrimitiveProps<AlertProps, 'div'>>;
+type PasswordFieldPrimitiveProps = PrimitiveProps<PasswordFieldProps, 'input'>;
+type ButtonPrimtiveProps = PrimitiveProps<ButtonProps, 'button'>;
```
Comment (ErikCH) [inline packages/react/src/components/AccountSettings/types.ts:15]:
```suggestion
type ButtonPrimitiveProps = PrimitiveProps<ButtonProps, 'button'>;
```

Then fix below as well

Comment (ErikCH) [reply packages/react/src/components/AccountSettings/types.ts:15]: This is an interesting pattern using the `PrimitiveProps` nice

**Review by calebpollman — APPROVED** [review]

File: packages/react/src/components/AccountSettings/types.ts (hunk 2 above)
Comment (wlee221) [reply packages/react/src/components/AccountSettings/types.ts]: Thanks for catching it, [37ded7e](https://github.com/aws-amplify/amplify-ui/pull/3131/commits/37ded7e14ac49da49152ba91488a57584b058d86)

**Pull request merged**